ros/ubuntu/Dockerfile

```diff
@@ -0,0 +1,65 @@
+## Provide Orocos Toolchain (RTT/OCL) on top of the official ROS docker images.
+ARG OROCOS_TOOLCHAIN_BRANCH=toolchain-2.9
+ARG ROS_DISTRO
+ARG UBUNTU_DISTRO
+
+## Stage deps: install additional Orocos Toolchain (with CORBA) dependencies
+FROM ros:${ROS_DISTRO}-ros-base-${UBUNTU_DISTRO} AS deps
+RUN apt-get update && apt-get install -y \
+    libboost-all-dev \
+    liblua5.1-0-dev \
+    libncurses5-dev \
+    libnetcdf-dev \
+    libreadline-dev \
+    libxml-xpath-perl \
+    libxml2-dev \
+    libeigen3-dev \
+    sip-dev \
+    libbullet-dev \
+    libangles-dev \
+    omniorb omniidl omniorb-idl omniorb-nameserver libomniorb4-dev \
+    ruby bundler \
+    $([[ "$UBUNTU_DISTRO" != "focal" ]] && echo "ruby-facets") \
+    && rm -rf /var/lib/apt/lists/*
+
+## Stage build: clone and build orocos_toolchain with colcon
+FROM deps AS build
+ARG OROCOS_TOOLCHAIN_BRANCH
+RUN git clone --recursive https://github.com/orocos-toolchain/orocos_toolchain.git -b ${OROCOS_TOOLCHAIN_BRANCH} /build/orocos_ws/src/orocos_toolchain \
+    && git clone --recursive https://github.com/orocos/orocos_kinematics_dynamics.git -b v1.4.0 /build/orocos_ws/src/orocos_kinematics_dynamics
```

Comment:
Nope. KDL is already preinstalled in `/opt/ros/${ROS_DISTRO}` and many packages depend on it. Installing a newer version from source on top will cause problems, although all 1.4 versions should be ABI-compatible.

For released packages, we should just install the debians. Building `orocos_toolchain` from source is a temporary solution only.

Reply:
If I am calling `. /opt/ros/${ROS_DISTRO}/setup.sh` and I still get `geometry/eigen_conversions` complaining for not finding `orocos_kdl`, does not that mean that KDL is not preinstalled in `/opt/ros/${ROS_DISTRO}` ?
```
CMake Error at geometry/eigen_conversions/CMakeLists.txt:4 (find_package):
  By not providing "Findorocos_kdl.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "orocos_kdl", but CMake did not find one.

  Could not find a package configuration file provided by "orocos_kdl" with
  any of the following names:

    orocos_kdlConfig.cmake
    orocos_kdl-config.cmake

  Add the installation prefix of "orocos_kdl" to CMAKE_PREFIX_PATH or set
  "orocos_kdl_DIR" to a directory containing one of the above files.  If
  "orocos_kdl" provides a separate development package or SDK, be sure it has
  been installed.
```